R/parse_lagtime.R

```diff
@@ -0,0 +1,20 @@
+#' Parse lagtime specified to main sim() function
+#' 
+#' @inheritParams sim
+#' 
+#' @returns a vector of character or numeric values
+#' 
+parse_lagtime <- function(
+  lagtime,
+  ode
+) {
+  if(!is.null(attr(ode, "lagtime")) && attr(ode, "lagtime")[1] != "undefined" && attr(ode, "lagtime")[1] != "NULL") {
+    if(is.null(lagtime)) { # only override from metadata if not specified by user
+      lagtime <- attr(ode, "lagtime")
+    }
+  }
+  if(is.null(lagtime)) {
+    lagtime <- c(0)
+  }
+  lagtime
```

Comment:
```suggestion
  lagtime_ode <- attr(ode, "lagtime")
  # override from ode if not specified by user and defined in ode
  if(is.null(lagtime) && !is.null(lagtime_ode) && !lagtime_ode %in% c("NULL", "undefined")) {
    lagtime <- lagtime_ode
  }
  if(is.null(lagtime)) {
    lagtime <- 0
  }
  lagtime
```

might be slightly more readable?